API/code/restapi/src/main/java/org/panalpina/api/rest/controller/AirFreightController.java

```diff
@@ -223,23 +223,27 @@ private String formatDate(String ETD) {
    * @return faFlightID
    * @throws IOException IOException
    */
-  private JsonNode parseFaFlightIDFromJson(String json, String date) throws IOException {
-    ObjectMapper objectMapper = new ObjectMapper();
-    JsonNode rootNode = objectMapper.readTree(json);
-    JsonNode flights = rootNode.get("FlightInfoStatusResult").get("flights");
-
-    if (flights.isArray()) {
-      for (final JsonNode objNode : flights) {
-        JsonNode faFlightID = objNode.get("faFlightID");
-        JsonNode filedDepartureTimeNode = objNode.get("filed_departure_time");
-        String filedDepartureDate = filedDepartureTimeNode.get("date").asText();
-        if (date.equals(filedDepartureDate)) {
-          return objNode;
-        }
-      }
-    }
-    return null;
-  }
+	private JsonNode parseFaFlightIDFromJson(String json, String date) throws IOException {
+		ObjectMapper objectMapper = new ObjectMapper();
+		JsonNode rootNode = objectMapper.readTree(json);
+		JsonNode flightInfoStatus = rootNode.get("FlightInfoStatusResult");
+		if (flightInfoStatus == null) {
+			return null;
```

Comment:
There should be DEBUG-level log statement indicating that `flightInfoStatus == null`.